src/Avalonia.Controls/Control.cs

```diff
@@ -329,5 +347,55 @@ protected override void OnKeyUp(KeyEventArgs e)
                 }
             }
         }
+
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == FlowDirectionProperty)
+            {
+                InvalidateFlowDirection();
+                
+                foreach (var visual in VisualChildren)
+                {
+                    if (visual is Control child)
+                    {
+                        child.InvalidateFlowDirection();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Computes the <see cref="IVisual.IsMirrorTransform"/> value according to the 
+        /// <see cref="FlowDirection"/> and <see cref="BypassFlowDirectionPolicies"/>
+        /// </summary>
+        public virtual void InvalidateFlowDirection()
+        {
+            FlowDirection thisFD = this.FlowDirection;
+            FlowDirection parentFD = FlowDirection.LeftToRight;
+
+            bool thisBypassFlowDirectionPolicies = BypassFlowDirectionPolicies;
```

Comment:
This isn't used

Reply:
I did not understand, later I use it:
`bool thisShouldBeMirrored = thisFD == FlowDirection.RightToLeft && !BypassFlowDirectionPolicies;`

Reply:
The initial value isn't used

Reply:
Ahh I see. Should I use it, or just remove it?

Reply:
Just move the declaration

Reply:
do you mean remove it?